### ISSUE TITLE
Deactivated DisplayRegion still renders when a second one is present

### ISSUE THREAD

**Issue by Schwarzbaer** [panda3d/panda3d#1682]
## Description
Deactivating the default display region works, unless then a second region is created, which reactivates rendering.

## Steps to Reproduce
```python
from direct.showbase.ShowBase import ShowBase


ShowBase()
base.camNode.get_display_region(0).active = False

# Scene
model = base.loader.load_model("models/environment")
model.reparent_to(base.render)
model.set_scale(0.25, 0.25, 0.25)
model.set_pos(-8, 42, 0)

# Additional DisplayRegion.
# Without these lines, only a grey window presents, as expected.
# With these lines, there'll be a camera view in the top-left, and the screenfilling default one will also be active again.
dr1 = base.win.make_display_region(0, 0.5, 0.5, 1)
dr1.set_sort(-1)
dr1.set_camera(base.make_camera(base.win))

# Done!
base.accept('escape', base.task_mgr.stop)
base.run()
```

## Environment
 * Operating system: Debian Linux
 * System architecture: x86_64
 * Panda3D version: 1.10.13
 * Installation method: pip
 * Python version (if using Python): 3.9.2

**Comment by niyazikemer** [comment]
`# TODO:
# 1. Test a logic that triggers when the boxes collide with the grass box.
# 1.1. Give more lifepower to the box agents
# 1.2. Create an offspring to the box for every 10 seconds that it survives.

# 2. Test the distance between the boxes and the grass box and if it was far, kill the box's brain.


from panda3d.core import loadPrcFileData
loadPrcFileData('', 'bullet-filter-algorithm groups-mask')
import torch
import sys
#import direct.directbase.DirectStart
from brain import Brain
from direct.showbase.DirectObject import DirectObject
from direct.showbase.InputStateGlobal import inputState

from panda3d.core import AmbientLight
from panda3d.core import DirectionalLight
from panda3d.core import Vec3
from panda3d.core import Vec4
from panda3d.core import Point3
from panda3d.core import TransformState
from panda3d.core import BitMask32

from panda3d.bullet import BulletWorld
from panda3d.bullet import BulletPlaneShape
from panda3d.bullet import BulletBoxShape
from panda3d.bullet import BulletRigidBodyNode
from panda3d.bullet import BulletDebugNode



#NK
from panda3d.core import WindowProperties
from panda3d.core import NodePath, Camera, PerspectiveLens
from direct.showbase.ShowBase import ShowBase
from panda3d.core import DisplayRegion#
from panda3d.core import GraphicsOutput

base = ShowBase()

# just one layer of a neural network of two neurons and two outputs




def calc_input(pos1,pos2):
    return torch.tensor([pos1,pos2])

class Game(DirectObject):

  def __init__(self):
    
    base.setBackgroundColor(0.1, 0.1, 0.8, 1)
    base.setFrameRateMeter(True)  
    base.setSceneGraphAnalyzerMeter(True)  
    
    base.cam.setPos(0, -20, 4)
    base.cam.lookAt(0, 0, 0)




    # Light
    alight = AmbientLight('ambientLight')
    alight.setColor(Vec4(0.5, 0.5, 0.5, 1))
    alightNP = render.attachNewNode(alight)

    dlight = DirectionalLight('directionalLight')
    dlight.setDirection(Vec3(1, 1, -1))
    dlight.setColor(Vec4(0.7, 0.7, 0.7, 1))
    dlightNP = render.attachNewNode(dlight)

    render.clearLight()
    render.setLight(alightNP)
    render.setLight(dlightNP)

    # Input
    self.accept('escape', self.doExit)
    self.accept('r', self.doReset)
    self.accept('f1', self.toggleWireframe)
    self.accept('f2', self.toggleTexture)
    self.accept('f3', self.toggleDebug)
    self.accept('f5', self.doScreenshot)

    self.accept('1', self.doSelect, [0,])
    self.accept('2', self.doSelect, [1,])
    self.accept('3', self.doSelect, [2,])
    self.accept('4', self.doSelect, [3,])
    self.accept('5', self.doSelect, [4,])
    self.accept('6', self.doSelect, [5,])

    inputState.watchWithModifiers('forward', 'w')
    inputState.watchWithModifiers('left', 'a')
    inputState.watchWithModifiers('reverse', 's')
    inputState.watchWithModifiers('right', 'd')
    inputState.watchWithModifiers('turnLeft', 'q')
    inputState.watchWithModifiers('turnRight', 'e')

    # Task
    taskMgr.add(self.update, 'updateWorld')

    
    # Physics
    self.setup()

  # _____HANDLER_____

  def doExit(self):
    self.cleanup()
    sys.exit(1)

  def doReset(self):
    self.cleanup()
    self.setup()

  def toggleWireframe(self):
    base.toggleWireframe()

  def toggleTexture(self):
    base.toggleTexture()

  def toggleDebug(self):
    if self.debugNP.isHidden():
      self.debugNP.show()
    else:
      self.debugNP.hide()

  def doScreenshot(self):
    base.screenshot('Bullet')

  def doSelect(self, i):
    self.selected_BoxNP = self.boxes[i]
    self.camNP.reparentTo(self.boxes[i].getChild(0))
  # ____TASK___

  def processInput(self, dt):
    force = Vec3(0, 0, 0)
    torque = Vec3(0, 0, 0)

    if inputState.isSet('forward'): force.setY( 1.0)
    if inputState.isSet('reverse'): force.setY(-1.0)
    if inputState.isSet('left'):    force.setX(-1.0)
    if inputState.isSet('right'):   force.setX( 1.0)
    if inputState.isSet('turnLeft'):  torque.setZ( 1.0)
    if inputState.isSet('turnRight'): torque.setZ(-1.0)
    
       
    # self_pos = self.boxNP.getPos()
    # self_pos1,self_pos2,self_pos3 = self_pos[0],self_pos[1],self_pos[2]
    # grass_pos = self.grassBox.getPos()
    # grass_pos1,grass_pos2,grass_pos3 = grass_pos[0],grass_pos[1],grass_pos[2]
    # fish_output = fish(torch.tensor([self_pos1,self_pos2,self_pos3,grass_pos1,grass_pos2,grass_pos3]))
    
    
    # force.setX(fish_output[0].item()*2)
    # force.setY(fish_output[1].item())
    # torque.setZ(fish_output[2].item())


    force *= 20.0
    torque *= 2

    force = render.getRelativeVector(self.selected_BoxNP, force)
    torque = render.getRelativeVector(self.selected_BoxNP, torque)

    self.selected_BoxNP.node().setActive(True)
    self.selected_BoxNP.node().applyCentralForce(force)
    self.selected_BoxNP.node().applyTorque(torque)



    ###### it is working
    # please refactor it to a function and check if it need setActive(True) every time
    for i in range(6):
      if self.boxes[i] != self.selected_BoxNP:
        selfpos = torch.tensor([self.boxes[i].getPos()[0],self.boxes[i].getPos()[1]])
        grasspos = torch.tensor([self.grassBox.getPos()[0],self.grassBox.getPos()[1]])
        inputs = torch.cat((selfpos,grasspos)) 
        fish_output = self.brains[i](inputs)
        fforce = Vec3(fish_output[0].item(),fish_output[1].item(),0)
        ttorque = Vec3(0,0,fish_output[2].item())
        fforce *= 20.0
        ttorque *= 2
        fforce = render.getRelativeVector(self.boxes[i], fforce)
        ttorque = render.getRelativeVector(self.boxes[i], ttorque)
        self.boxes[i].node().setActive(True)
        self.boxes[i].node().applyCentralForce(fforce)
        self.boxes[i].node().applyTorque(ttorque) 



  def update(self, task):
    dt = globalClock.getDt()

    self.processInput(dt)
    #self.world.doPhysics(dt)
    self.world.doPhysics(dt, 5, 1.0/180.0)
    
    #self.collisionDetector()

    return task.cont

  def cleanup(self):
    self.world.removeRigidBody(self.groundNP.node())
    self.world.removeRigidBody(self.boxNP.node())
    self.world = None

    self.debugNP = None
    self.groundNP = None
    self.boxNP = None

    self.worldNP.removeNode()

  def setup(self):
    
    self.worldNP = render.attachNewNode('World')
    
    # World
    self.debugNP = self.worldNP.attachNewNode(BulletDebugNode('Debug'))
    self.debugNP.show()
    self.debugNP.node().showWireframe(True)
    self.debugNP.node().showConstraints(True)
    self.debugNP.node().showBoundingBoxes(False)
    self.debugNP.node().showNormals(True)
    

    #self.debugNP.showTightBounds()
    #self.debugNP.showBounds()
    # THIS makes the collision detection Fald
    

    self.world = BulletWorld()
    # I cant make the collision detection groups work.it just didnt work so use not groups but individual masks
    
    self.world.setGravity(Vec3(0, 0, -9.81))
    self.world.setDebugNode(self.debugNP.node())

    # Ground (static)
    shape = BulletPlaneShape(Vec3(0, 0, 1), 1)

    self.groundNP = self.worldNP.attachNewNode(BulletRigidBodyNode('Ground'))
    self.groundNP.node().addShape(shape)
    self.groundNP.setPos(0, 0, -2)
    self.groundNP.setCollideMask(BitMask32.bit(0))

    self.world.attachRigidBody(self.groundNP.node())
    
    
    # Boxes (dynamic)
    self.boxes = [None,]*6
    self.brains = {}
    
    for i in range(6):
      shape = BulletBoxShape(Vec3(0.5, 0.5, 0.5))
      boxNP = self.worldNP.attachNewNode(BulletRigidBodyNode('Box'))
      boxNP.node().setMass(1.0)
      boxNP.node().addShape(shape)
      boxNP.setPos(i + 3, i + 3, 2)
      self.world.attachRigidBody(boxNP.node())    
      boxNP.setCollideMask(BitMask32.bit(1))

      visualNP = loader.loadModel('models/box.egg')
      visualNP = loader.loadModel('minecraft_blocks/sand-block.glb')      
      visualNP.setScale(0.5)
      visualNP.clearModelNodes()
      visualNP.reparentTo(boxNP)
      self.brains[i] = Brain()
      self.boxes[i] = boxNP
    self.selected_BoxNP = self.boxes[0]
    print(boxNP.getChild(0))

    #THE GRASS BOX
    shape = BulletBoxShape(Vec3(0.5, 0.5, 0.5))
    self.grassBox = self.worldNP.attachNewNode(BulletRigidBodyNode('grassBox'))
    #self.grassBox.setCollideMask(BitMask32.bit(1))
    self.grassBox.setCollideMask(BitMask32.bit(0))
    # self.grassBox.node().setMass(1.0)
    self.grassBox.node().addShape(shape)
    self.grassBox.setPos(0, 0, -0.5)
    
    #self.grassBox.setScale(2, 1, 0.5)
    
    
    #self.grassBox.node().setDeactivationEnabled(False)

    self.world.attachRigidBody(self.grassBox.node())

    

    visualgrassBoxNP = loader.loadModel('minecraft_blocks/grass-block.glb')
    visualgrassBoxNP.setScale(0.5)
    visualgrassBoxNP.clearModelNodes()
    visualgrassBoxNP.reparentTo(self.grassBox)
    
    # Bullet nodes should survive a flatten operation!
    #self.worldNP.flattenStrong()
    #render.ls()

    # wp = WindowProperties()
    # wp.setSize(960, 540)
    # wp.title = 'Second'
    # win2 = base.openWindow(makeCamera=True,props=wp, aspectRatio=16/9,name = "hi")
    # print(win2.getNumDisplayRegions())
    # print(win2.getDisplayRegions())
    # print(win2.active_display_regions)
    # win2.active_display_regions[0].setCamera(camNP)



    lens = PerspectiveLens()
    lens.setFov(100)
    my_cam = base.makeCamera(base.win, sort=0,camName="x_cam")
    my_cam.setPos(0, -0.5, 2)
    #my_cam.setHpr(1, 100, 100)
    my_cam.node().setLens(lens)
    my_cam.node().setActive(True)
    
    self.camNP = NodePath(my_cam)
    self.camNP.reparentTo(self.boxes[0].getChild(0))  
    region2=base.win.makeDisplayRegion(0, 1, 0, 1)
    region2.setCamera(self.camNP)
    region2.setSort(2)
    

    # print(base.win.getNumDisplayRegions())
    # print(base.win.getDisplayRegions())
    # print(base.win.active_display_regions)


    self.world.setGroupCollisionFlag(1, 1, False)
    self.world.setGroupCollisionFlag(1, 0, True)

    # wp = WindowProperties()
    # wp.setSize(960, 540)
    # wp.title = 'Second'
    # win2 = base.openWindow(makeCamera=False,props=wp, aspectRatio=16/9,name = "hi")
    # dr2 = win2.makeDisplayRegion(0.5, 1, 0.5, 1)

    pilotcam = base.makeCamera(base.win, sort=1,camName="p_cam")
    self.pilotcamNP = NodePath(pilotcam)
    pilotcam.setPos(0, 0, 5)
    pilotcam.node().setLens(lens)
    pilotcam.setHpr(0, -90, 0)
    pilotcam.node().setActive(True)
    pilotcamNP.display_regions
    


    

    region=base.win.makeDisplayRegion(0, 0.25, 0, 0.25)
    # if it is active or not set its default value is True
    base.cam.node().setActive(False)

    region.setCamera(self.pilotcamNP)
    region.setSort(2)


      


# this is contact detector.
   
  def collisionDetector(self):
    #result = self.world.contactTest(self.grassBox.node())
    result = self.world.contactTestPair(self.grassBox.node(), self.boxNP.node())

    print(result.getNumContacts())

    for contact in result.getContacts():
      print(contact.getNode0())
      print(contact.getNode1())

      mpoint = contact.getManifoldPoint()
      
      print(mpoint.getDistance())
      print(mpoint.getAppliedImpulse())
      print(mpoint.getPositionWorldOnA())
      print(mpoint.getPositionWorldOnB())
      print(mpoint.getLocalPointA())
      print(mpoint.getLocalPointB())

game = Game()
base.run()`

**Comment by rdb** [comment]
The original region is still inactive, the problem is that `base.make_camera` also creates a new display region, set up to render the main scene, which covers the screen by default.

You instead want to let base.make_camera just create a DR for you:
```python
base.make_camera(base.win, displayRegion=(0, 0.5, 0.5, 1), sort=-1)
```

Or just create your own camera, without using make_camera:
```python
dr1 = base.win.make_display_region(0, 0.5, 0.5, 1)
dr1.set_sort(-1)
dr1.set_camera(base.render.attach_new_node(Camera("myNewCamera")))
```

**Comment by niyazikemer** [comment]
Wow OK, Thank you.
